metrics/collector.go

```diff
@@ -143,6 +144,9 @@ func NewCollector(logger zerolog.Logger) Collector {
 func registerMetrics(logger zerolog.Logger, metrics ...prometheus.Collector) error {
 	for _, m := range metrics {
 		if err := prometheus.Register(m); err != nil {
+			if errors.As(err, &prometheus.AlreadyRegisteredError{}) {
```

Comment:
this seems fine, but also points to a bug where we're registering the same metrics more than once. any idea why you needed to add this?

Reply:
The EVM GW can be bootstrapped again & again during E2E tests, so the metrics were registered more than once.